subsys/bluetooth/host/sdp.c

```diff
@@ -1117,6 +1175,11 @@ static uint16_t sdp_svc_att_req(struct bt_sdp *sdp, struct net_buf *buf,
 		}
 
 		state.last_att = net_buf_pull_u8(buf) + 1;
+		state.last_att_offset = net_buf_pull_le16(buf);
```

Comment:
Shouldn't that be `pull_be16()` since SDP is a big-endian protocol? Did you actually test that this code works?

Reply:
It has no relation with the SDP endian. this field is the data of ContinuationState. A send this field data to B, and B send it back without any parsing, then B parse this data and know the data offset. I did simple test with BT_SDP_SERVICE_NAME that is longer than SDP_MTU.